### PR TITLE
refactor: code cleanups

### LINKED ISSUE #60: chore: top of waterfall 🌊
### Background
Many PR comments for the new dockerised version are better addressed at the end of the PR waterfall. This issue tracks those.

### Details
All PR action items best suited for top of water fall are integrated.

### Acceptance criteria

- [x] refactor: modes to commands with 1:1 mapping
- [x] refactor: [properly export types](https://github.com/cowprotocol/tenderly-watch-tower/pull/39#discussion_r1336078118)
- [x] refactor: [separate chain watcher](https://github.com/cowprotocol/tenderly-watch-tower/pull/47#discussion_r1336144485)
- [x] chore: [script to run cli](https://github.com/cowprotocol/tenderly-watch-tower/pull/39#discussion_r1336068755)
- [x] #67
- [x] chore: [parameterize db location](https://github.com/cowprotocol/tenderly-watch-tower/pull/52#discussion_r1334347541)
- [x] docs: jsdoc exported interfaces, eg [here](https://github.com/cowprotocol/tenderly-watch-tower/pull/39#discussion_r1336073076)
- [x] docs: update readme
- [x] docs: docker usage

### PR DESCRIPTION
**Pull request by mfw78**
# Description
As was identified in multiple preceding PRs, some things could be separated better. Notably there was a universal desire to extract the `ChainContext` to it's own file. 

# Changes

This PR contains some of the work required to address the chore task list in #60.

- [x] refactor: modes to commands with 1:1 mapping
- [x] refactor: properly export types
- [x] refactor: separate chain watcher
- [x] refactor: `lastResult` -> `lastHint` for nomenclature
- [x] docs: jsdoc exported interfaces
- [x] docs: readme

## How to test

```
yarn
yarn ts-node ./src/index.ts run --rpc http://10.23.0.2:8545 --rpc http://10.23.0.2:8645 --deployment-block 17883049 --deployment-block 29389123 --page-size 0
```

**NOTE**: Replace the above RPCs as required.

## Related Issues

Fixes #60

### CODE REVIEW

File: src/domain/checkForAndPlaceOrder.ts [hunk 1]
```diff
@@ -523,21 +528,15 @@ function _handleOrderBookError(
...
+    if (BACKPRESSURE_API_ERRORS.includes(data?.errorType)) {
+      const nextPollTimestamp = blockTimestamp + BACKPRESSURE_DELAY;
       return {
         result: PollResultCode.TRY_AT_EPOCH,
         epoch: nextPollTimestamp,
         reason: `Not enough allowance/balance (${
           data?.errorType
         }). Scheduling next polling in ${Math.floor(
-          WAITING_TIME_SECONDS_FOR_NOT_BALANCE / 60
+          BACKPRESSURE_DELAY / 60
```
Comment (anxolin) [inline src/domain/checkForAndPlaceOrder.ts:539]: shoudn't be `BACK_REASSURE_DELAY`? 
Also, i think this name is too generic. Before it was informing what is this waiting for. this is just reducing the checks in the case the owner don't have balance

Now it looks like its a delay to adjust for a producer consumer mismatch. why this change?

File: src/domain/checkForAndPlaceOrder.ts [hunk 2]
```diff
@@ -109,32 +114,32 @@ async function _checkForAndPlaceOrder(
       const logPrefix = `[checkForAndPlaceOrder::${orderRef}]`;
       const logOrderDetails = `Processing order from TX ${conditionalOrder.tx} with params:`;
 
-      const { result: lastResult } = conditionalOrder.pollResult || {};
+      const { result: lastHint } = conditionalOrder.pollResult || {};
```
Comment (anxolin) [inline src/domain/checkForAndPlaceOrder.ts:117]: Just curious, why `lastHint` for the `last poll result`?

Comment (mfw78) [reply src/domain/checkForAndPlaceOrder.ts:117]: Sure, sorry I should have communicated my thinking behind this. It seems that the _outcome_ of the polling is to _hint_ to the watch-tower what should be done. This makes it easier, IMO, when reasoning.

IMO this then makes it easier when looking at comparisons, such as:

```typescript
      if (
        lastHint?.result === PollResultCode.TRY_AT_EPOCH &&
        blockTimestamp < lastHint.epoch
      ) {
```

Having some "meaning" that's built into the name, as opposed to contextual, I think makes easier reading here?

File: src/domain/checkForAndPlaceOrder.ts (hunk 1 above)
Comment (mfw78) [reply src/domain/checkForAndPlaceOrder.ts:539]: The idea here is that there may be other API errors of which we want to back-off hitting the API. Maybe `BACKOFF_DELAY` would be more appropriate?

**Review by alfetopito — APPROVED** [review]

File: src/domain/checkForAndPlaceOrder.ts (hunk 1 above)
Comment (anxolin) [reply src/domain/checkForAndPlaceOrder.ts:539]: not too strong oppinion, but at least would be good to mention is for the API errors, I wouldn't know what `BACKOFF_DELAY` is if i see it. And this is only used also for errors related to balance and allowance. I checked the other errors, and for now they didn't make sense to me to do the same treatment. 

If there's other errors that need to add some delay, is not clear it will be the same time, or same treatment. So maybe we are just getting ahead of ourselves. 

anyways, fine with me either way

File: src/domain/checkForAndPlaceOrder.ts (hunk 2 above)
Comment (anxolin) [reply src/domain/checkForAndPlaceOrder.ts:117]: okok, it also works

**Review by anxolin — APPROVED** [review]

File: src/domain/checkForAndPlaceOrder.ts (hunk 1 above)
Comment (mfw78) [reply src/domain/checkForAndPlaceOrder.ts:539]: Upon looking at this for a second time, I unified the configuration and typings for api errors that have specifiable / configurable delays, and api errors that should trigger try again in next block. I think it's in a bit of a more readable state now :smiley:
